pkg/registry/batch/job/strategy.go

```diff
@@ -147,10 +178,28 @@ func (jobStrategy) Validate(ctx context.Context, obj runtime.Object) field.Error
 	if job.Spec.ManualSelector == nil || *job.Spec.ManualSelector == false {
 		generateSelector(job)
 	}
-	opts := pod.GetValidationOptionsFromPodTemplate(&job.Spec.Template, nil)
+	opts := validationOptionsForJob(job, nil)
 	return validation.ValidateJob(job, opts)
 }
 
+func validationOptionsForJob(newJob, oldJob *batch.Job) validation.JobValidationOptions {
+	var newPodTemplate, oldPodTemplate *core.PodTemplateSpec
+	if newJob != nil {
+		newPodTemplate = &newJob.Spec.Template
+	}
+	if oldJob != nil {
+		oldPodTemplate = &oldJob.Spec.Template
+	}
+	opts := validation.JobValidationOptions{
+		PodValidationOptions:    pod.GetValidationOptionsFromPodTemplate(newPodTemplate, oldPodTemplate),
+		AllowTrackingAnnotation: utilfeature.DefaultFeatureGate.Enabled(features.JobTrackingWithFinalizers),
+	}
+	if oldJob != nil {
+		opts.AllowTrackingAnnotation = opts.AllowTrackingAnnotation && hasJobTrackingAnnotation(oldJob)
```

Comment:
should be `||`?

Reply:
And that is true with this check.

The reason why we require that the old job also has the annotation is that we don't support migrating from a legacy tracking to the new tracking.

Reply:
IIUC, if I disable the feature, and try to suspend a job which already has the annotation, I will get a "restricted annotation" error, right?